File: redwood-yoga-java/src/main/java/io/github/orioncraftmc/meditate/interfaces/YogaMeasureFunction.java
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package io.github.orioncraftmc.meditate.interfaces;

import io.github.orioncraftmc.meditate.YogaNode;
import io.github.orioncraftmc.meditate.enums.YogaMeasureMode;
import io.github.orioncraftmc.meditate.internal.YGSize;

public interface YogaMeasureFunction {
    /**
     * Return a value created by YogaMeasureOutput.make(width, height);
     */
    YGSize measure(
            YogaNode node,
            float width,
            YogaMeasureMode widthMode,
            float height,
            YogaMeasureMode heightMode);

}

File: redwood-yoga/src/commonMain/kotlin/app/cash/redwood/yoga/interfaces/YogaMeasureFunction.kt
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
package app.cash.redwood.yoga.interfaces

import app.cash.redwood.yoga.enums.YogaMeasureMode
import app.cash.redwood.yoga.YogaNode
import app.cash.redwood.yoga.internal.YGSize

interface YogaMeasureFunction {
    fun measure(
      node: YogaNode,
      width: Float,
      widthMode: YogaMeasureMode,
      height: Float,
      heightMode: YogaMeasureMode
    ): YGSize
}
